activemodel/lib/active_model/secure_password.rb

```diff
@@ -36,8 +36,11 @@ module ClassMethods
       # validation will fail.
       #
       # All of the above validations can be omitted by passing
-      # <tt>validations: false</tt> as an argument. This allows complete
-      # customizability of validation behavior.
+      # <tt>validations: false</tt> as an argument.
+      # Alternatively you can pass a Hash to +:validations+ with the standard
+      # +:if+ / +:unless+ / +:on+ / +:prepend+ keys. (See
+      # ActiveModel::Validations::ClassMethods#validate for more information.)
+      # This allows complete customizability of validation behavior.
```

Comment:
I realize `:prepend` is accepted, but I don't think we should document it.  In fact, I considered writing `validations.slice(:if, :unless, :on)`, but decided to keep the code simple instead.

I don't think there is a strong use case for supporting it.  Furthermore, it could complicate adding more validations that use `validation_options` in the future.  Now, if someone were to specify `prepend: true`, the password presence validation would be added to the top of validations list.  In the future, if we add another `validate(validation_options)` block at the bottom of `has_secure_password`, then `prepend: true` would put *that* validation at the top of the list, breaking the expectation.

Given this and my suggestion regarding `password_challenge`, I would prefer to use the wording from https://github.com/rails/rails/pull/45487#pullrequestreview-1026441129: 

```ruby
      # The password presence validation can be conditionally enforced by 
      # passing an options hash to +:validations+ with the standard +:if+ / 
      # +:unless+ / +:on+ keys. (See ActiveModel::Validations::ClassMethods#validates 
      # for more information.)  Alternatively, all of the above validations can 
      # be omitted by passing <tt>validations: false</tt>. This allows complete 
      # customizability of validation behavior.
```

(Note: "This allows complete customizability of validation behavior" refers specifically to `validations: false` rather than the `:validations` option as a whole, so I am trying to keep those sentences adjacent.)